### DISCUSSION TITLE
Keeping my models up to date with changes made during migrations

### DISCUSSION THREAD

**Discussion by kiwavi** [sqlalchemy/sqlalchemy#11012] [Usage Questions] [answered]
Hi guys. So I'm using fastAPI for my backend and alembic for migrations. Is there a way to automatically update my models file (sqlalchemy) once I make changes via migrations. For example, I've added a field to a table, applied the migrations, but my models file do not reflect the changes - the new field. Thanks in advance.

**Comment by CaselIT** [comment]
Hi,

I'm assuming you are not using autogenerate of alembic, but creating the migration manually correct?

**Reply by kiwavi** [reply]
Hello. Yes. Followed the tutorial and I'm running the migrations manually. Tried looking into autogenerate and I grasped that it usually compares your metadata to the database.

**Reply by CaselIT** [reply, marked as answer]
Ok, so I don't think there is anything automated that from a migration will update your models python files.

To write the models from the database schema you can look into sqlacodegen, but I believe that's more targeted to one-off generation given an existing database. Also I think it will generate all models not just update the existing ones.

The usual use case with alembic would be to modify the models files, then create the migration using alembic autogenerate to create a candidate migration, that can be manually reviewed.

**Reply by kiwavi** [reply]
Yes the use case with alembic is what I need. Thanks for the clarification man.
